### PR TITLE
update docstrings to new format

### PR DESCRIPTION
**Pull request by RatanRSur**
Shamelessly took the low hanging fruit :)

### CODE REVIEW

**Comment by ararslan** [comment]
Thanks for the contribution!

As an aside, it's best not to use the `@` symbol at all in commit messages, since it actually ends up sending a notification to the user with that name. In this case, [this person](https://github.com/doc) has gotten or will get a notification. I think the common practice is to say `at-doc` or `doc macro`, etc. in commit messages. If you'd like, you can `git commit --amend` and then force push to your branch.

So that GitHub will auto-close the issue once this is merged: Fixes #159

File: src/broadcast.jl [hunk 1]
```diff
@@ -68,8 +68,8 @@ if VERSION < v"0.5.0-dev+5434"
     end
 
     @eval let cache = Dict{Any, Dict{Bool, Dict{Int, Dict{Int, Any}}}}()
-        @doc """
-        `broadcast!(f, B::NullableArray, As::NullableArray...; lift::Bool=false)`
+        """
+            `broadcast!(f, B::NullableArray, As::NullableArray...; lift::Bool=false)`
```
Comment (ararslan) [inline src/broadcast.jl]: When these are indented, the backticks should be removed.

**Review by ararslan — CHANGES_REQUESTED** [review]

**Comment by RatanRSur** [comment]
Oh, cool I did not know that about @'s. Thanks!

Let me know if there are more changes.

File: src/indexing.jl [hunk 2]
```diff
@@ -129,8 +129,8 @@ function Base.to_index(X::NullableArray)
     Base.to_index(X.values)
 end
 
-@doc """
-`nullify!(X::NullableArray, I...)`
+"""
+   nullify!(X::NullableArray, I...)
 
 This is a convenience method to set the entry of `X` at index `I` to be null
 """ ->
```
Comment (ararslan) [inline src/indexing.jl]: The arrows at the end of the doc strings should also go; they aren't part of the new syntax.

Comment (RatanRSur) [reply src/indexing.jl]: Done, also I noticed I messed up and used 3 space indents so I fixed that as well

**Review by ararslan — APPROVED** [review]

**Comment by ararslan** [comment]
@davidagold Is the nightly failure a known issue?

File: src/broadcast.jl [hunk 3]
```diff
@@ -68,8 +68,8 @@ if VERSION < v"0.5.0-dev+5434"
     end
 
     @eval let cache = Dict{Any, Dict{Bool, Dict{Int, Dict{Int, Any}}}}()
-        @doc """
-        `broadcast!(f, B::NullableArray, As::NullableArray...; lift::Bool=false)`
+        """
+            broadcast!(f, B::NullableArray, As::NullableArray...; lift::Bool=false)
         This method implements the same behavior as that of `broadcast!` when called on
```
Comment (nalimilan) [inline src/broadcast.jl:74]: Missing line break here.

File: src/primitives.jl [hunk 4]
```diff
@@ -112,30 +112,30 @@ function Base.reshape(X::NullableArray, dims::Dims) # -> NullableArray
...
+    endof(X::NullableArray)
 
 Returns the last entry of `X`.
-""" ->
+"""
 Base.endof(X::NullableArray) = endof(X.values) # -> Int
 
-@doc """
-
-""" ->
```
Comment (nalimilan) [inline src/primitives.jl:138]: This empty comment is weird, probably time to remove it.

File: src/typedefs.jl [hunk 5]
```diff
@@ -13,11 +13,11 @@
 #
 # TODO: Ensure that size(values) == size(isnull) using inner constructor.
 # TODO: Implement outer constructor required once we add an inner constructor.
-@doc """
-`NullableArray{T, N}` is an efficient alternative to `Array{Nullable{T}, N}`.
+"""
+    NullableArray{T, N} is an efficient alternative to `Array{Nullable{T}, N}`.
```
Comment (nalimilan) [inline src/typedefs.jl]: This line isn't a signature, it's just a sentence, so keep it as it was previously.

**Review by nalimilan — CHANGES_REQUESTED** [review]
Thanks! Looks good apart from a few details.

**Comment by nalimilan** [comment]
@ararslan The nightly failure is known, that's a problem with `broadcast`.

File: src/typedefs.jl (hunk 5 above)
Comment (ararslan) [reply src/typedefs.jl]: Thanks for the thorough review, @nalimilan, and for catching some stuff I missed!

**Review by nalimilan — APPROVED** [review]

**Comment by nalimilan** [comment]
Woops, completely forgot about this. It's no longer October, but let's pretend it is...

**Pull request merged**